oat-sa/extension-parcc-tei
# Fix/TR-3424/graphic PCIs in test review

Related to: https://oat-sa.atlassian.net/browse/TR-3424

Depends on:

- [x] https://github.com/oat-sa/qti-sdk/pull/339 - `fix/TR-3424/unmarshall-list-inside-record-container`
- [ ] taoOutcomeUi - `feature/TR-3424/handle-record-type-response-variable`
- [ ] lib-tao-qti - `feature/TR-3424/transform-record-to-database-store`

Related issue: https://github.com/oat-sa/extension-tao-itemqti/pull/2255 - popups in PCI authoring

Fix review mode for these PCI: update `destroy` and `getInstance` function.
**Root of issue:** in review we render PCI 2 times, one initial render, then destroy and render with state again. `Destroy` removed all markup - and second render fails because no elements in DOM. Also was no call to `setResponse` in `getInstance` function.

Also was updated responses record for some of the PCI, to have proper structure.

**How to test:**

1. install extension
2. switch `taoOutcomeUi` and 2 libs `vendor/tao-sa/lib-tao-qti` and `vendor/qtism/qtism` to proper branches
3. create items with all PCIs (6 items)
4. create test
5. create delivery
6. pass delivery
7. go to Results page
8. view delivery execution, check that all Responses set in items Review (click Review button)

![image](https://user-images.githubusercontent.com/25976342/197821058-b3073d6f-e783-48ea-8c20-093ea460e653.png)

![image](https://user-images.githubusercontent.com/25976342/197821001-dc62799e-2c2c-48d0-a39a-f90218bdc235.png)
